keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,572 @@
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a user submitting workloads, I want to easily see the status of an entire Workload,
+or check on why my workload is not being admitted. I can do this with the new proposed plugin.
+
+#### Story 2
+
+As a user coming from High Performance Computing, I am not comfortable with using
+`kubectl` and don't want to learn an entirely new means to interact with workloads.
+The proposed plugin makes the transition much easier for me.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+Maintaining the plugin in-sync with the latest version of Kueue will add additional
+maintainer responsibilities.
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Summary
+
+Kueuectl is a command-line tool for interacting with Kueue, and can serve as a standalone tool or be renamed/installed to act as a kubectl plugin. Using `kueuectl` a user can manage workloads. An alternative (shorter and easier to type name) might also be `kuctl`, which I'll use for the remainder of this document. Thus, the main client takes the following format:
+
+```bash
+kuctl [subcommand] [flags]
+```
+
+As a kubectl plugin, we would install it named as `kubectl-kueue` and then the interaction would be:
+
+```bash
+kubectl kueue [subcommand] [flags]
+```
+
+We can start with basic query of workload metadata and status, and move toward a tool that can further create / delete or otherwise interact with workloads. This design document will proceed with proposed interactions and example tables, which would be printed in the terminal.
+
+
+### Version
+
+```bash
+kuctl version
+```
+
+### Listing Workloads
```

Comment:
I think it would be useful if this would work with the job name. And describe would make an aggregation of the job+workload objects, so that users can have a single view of everything that is happening with their job.

With that in mind, I opened #992

Reply:
Going with job-name will require to  either specify the job type also , or try to list all the supported types objects having that name.

Reply:
yes, we need to include the API group and kind